fdbserver/KeyValueStoreRocksDB.actor.cpp

```diff
@@ -2492,6 +2493,48 @@ struct RocksDBKeyValueStore : IKeyValueStore {
 		return EncryptionAtRestMode(EncryptionAtRestMode::DISABLED);
 	}
 
+	Future<Void> ingestSSTFiles(std::string bulkLoadLocalDir,
+	                            std::shared_ptr<BulkLoadFileSetKeyMap> localFileSets) override {
+		// Create a list of SST files to ingest
+		std::vector<std::string> sstFiles;
+		for (const auto& [range, fileSet] : *localFileSets) {
+			if (fileSet.hasDataFile()) {
+				sstFiles.push_back(fileSet.getDataFileFullPath());
+			}
+		}
+
+		// Configure ingestion options
+		rocksdb::IngestExternalFileOptions options;
+		options.move_files = true; // Move files instead of copying
+		options.snapshot_consistency = false;
+		options.allow_global_seqno = false;
+		options.allow_blocking_flush = false;
+		options.verify_checksums_before_ingest = true; // Verify checksums before ingestion
```

Comment:
Do we need to setup anything particularly when creating the sst files?

Reply:
It's a rocksdb thing -- checksums on blocks inside the SST. Nothing for us to do on our part. The flag says whether to check before making the data live. Restore sets it to true.  sharded rocks has it as a command-line option.